### PR TITLE
fix: [DEL-4911]: new field for immutable delegate published version in ring

### PR DESCRIPTION
**Pull request by AnupamIO**
https://harness.atlassian.net/browse/DEL-4911

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37825)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37825)

**Comment by AnupamIO** [comment]
trigger messagecheck

File: 400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java [hunk 1]
```diff
@@ -812,9 +812,15 @@ public DelegateStatus getDelegateStatusWithScalingGroups(String accountId) {
                 : delegateVersionListWithoutPatch(delegateConfiguration.getDelegateVersions()))
         .scalingGroups(scalingGroups)
         .delegates(buildInnerDelegates(accountId, delegatesWithoutScalingGroup, activeDelegateConnections, false))
+        .publishedImmutableDelegateVersion(
+            FetchDelegateVersionFromImage(delegateVersionService.getImmutableDelegateImageTag(accountId)))
         .build();
   }
 
+  private String FetchDelegateVersionFromImage(String immutableDelegateImageTag) {
```
Comment (xingchi-jin) [inline 400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java:820]: @NotBlank final String immutableDelegateImageTag

**Review by AnupamIO — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37825)* status: 3 file(s) to review. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 820 at r1](https://reviewable.io/reviews/harness/harness-core/37825#-NCaqjSY0gAOpFsivmau-r1-820:-NCcj7b44MfulK1CN1XC:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/791231b7a5059d41812d9d4a6a42bb610b9f91e7/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L820)):*
<details><summary><i>Previously, xingchi-jin (Xingchi Jin) wrote…</i></summary><blockquote>

@NotBlank final String immutableDelegateImageTag
</blockquote></details>

Done.

**Pull request merged**